### ISSUE TITLE
version update from 1.6.20-dev1673 -> 1.7.0-dev1686 breaks AnimatedContent

### ISSUE THREAD

**Issue by Zenturas** [JetBrains/compose-multiplatform#5008]
**Describe the bug**
I have my custom navigation with screen animations using AnimatedContent.
Before updating to 1.7.0-xxx there was a nice transition. 
After updating it happens nothing for the "animation duration" and after that the next screen appears.
See videos.

Android not affected.

**Affected platforms**
- Desktop (Windows, Linux, macOS)

**Versions**
- Libraries:
- Compose Multiplatform version: 1.7.0-dev1686
- Kotlin version: 2.0
- OS version(s) (required for Desktop and iOS issues): win11
- JDK (for desktop issues): 19

**To Reproduce**
I have no reproducer just a snippet from my code but its simple
```
@OptIn(ExperimentalAnimationApi::class)
	@Composable
	internal fun invokeScreen(
		modifier: Modifier,
		forwardEnterTransition: EnterTransition,
		forwardExitTransition: ExitTransition,
		backEnterTransition: EnterTransition,
		backExitTransition: ExitTransition
	) {
		AnimatedContent(
			transitionSpec = {
				when (direction) {
					ScreenDirection.NEUTRAL -> forwardEnterTransition togetherWith forwardExitTransition
					ScreenDirection.FORWARD -> forwardEnterTransition togetherWith forwardExitTransition
					ScreenDirection.BACK -> backEnterTransition togetherWith backExitTransition
				}
			},
			targetState = currentScreen,
			modifier = modifier,
			content = {
				if (transition.currentState == transition.targetState)
					direction = ScreenDirection.NEUTRAL
				it.invoke() // it = the screen (just a composable function)
			}
		)
	}
```

**Gifs**

### Before updating
![beforeupdate](https://github.com/JetBrains/compose-multiplatform/assets/128176107/9978bbb4-1e1c-41ba-99b3-0c3792fa6c27)

### After updating
![afterupdate](https://github.com/JetBrains/compose-multiplatform/assets/128176107/6a055f3c-9596-4730-a332-5667a0677acd)

**Comment by MatkovIvan** [comment]
Known and tricky one. It's supposed to be fixed in https://github.com/JetBrains/compose-multiplatform-core/pull/1412

**Comment by MatkovIvan** [comment]
Should be fixed in 1.7.0-dev1698

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5008">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
